### DISCUSSION TITLE
Dumb v3.0.0 API Question - I'm probably not reading the docs right

### DISCUSSION THREAD

**Discussion by Webreaper** [SixLabors/ImageSharp#2380] [Q&A] [answered]
So currently using v2 I have [this](https://github.com/Webreaper/Damselfly/blob/master/Damselfly.Core.ImageProcessing/ImageSharpProcessor.cs#L160):
```
    // Load the image, getting the source format
    using var img = await Image.LoadAsync(input, out var format);

    // ... now do some some funky stuff... 

    // Now save - in the same format as the original, but a different path
    await img.SaveAsync(output, format);
```
This ensures the new/transformed file is generated in the same format as the original - JPEG, PNG whatever.

In v3.0.0 the API with the output param for the format seems to be missing (or I can't find it). How do I do this now? `SaveAsync` to a stream requires a `format` or `decoder` parameter, which I now don't have.

```
    DecoderOptions options = new() { TargetSize = new( width: config.MaxImageSize, height: config.MaxImageSize ) };

    // There's no overload for this method that takes output param to return the format
    using var img = await Image.LoadAsync(options, input);

    // ... now do some some funky stuff... 

    // Doesn't work, because I now don't have a format or decoder to pass it
    await img.SaveAsync( outputStream ); 
```

**Comment by JimBobSquarePants** [comment, marked as answer]
Yeah, we've got a breaking change. 

The whole `out` API was clunky and wasn't consistant across sync/async so we did a complete overhaul of the Load APIs to simplify things.

You can now find that information in the image metadata

```c#
DecoderOptions options = new() { TargetSize = new( width: config.MaxImageSize, height: config.MaxImageSize ) };
using var img = await Image.LoadAsync(options, input);

// ... now do some some funky stuff... 

await img.SaveAsync(outputStream,  image.Metadata.DecodedImageFormat); 
```

https://docs.sixlabors.com/articles/imagesharp/imageformats.html#loading-and-saving-specific-image-formats

**Comment by Webreaper** [comment]
Perfect, thanks. I thought it was somewhere in there, but couldn't spot it. Much cleaner API - good work!

**Reply by JimBobSquarePants** [reply]
Thanks!
